File: src/com/esielkar/flyweight/Context.java
package com.esielkar.flyweight;

public class Context {
    private int uniqueState;
    private Flyweight flyweight;

    public Context(int repeatingState, int uniqueState) {
        this.uniqueState = uniqueState;
        flyweight = FlyweightFactory.getFlyweight(repeatingState);
    }

    public void operation() {
        flyweight.operation(uniqueState);
    }

    public boolean hasFlyweight(Flyweight flyweight) {
        return this.flyweight == flyweight;
    }

    @Override
    public String toString() {
        return "Context: uniqueState = " + uniqueState + " " + flyweight;
    }

}

File: dart/design_patterns/bin/flyweight/context.dart
import 'flyweight.dart';

class Context {
  final int _uniqueState;
  final Flyweight _flyweight;

  Context(int repeatingState, int uniqueState)
      : _uniqueState = uniqueState,
        _flyweight = FlyweightFactory.getFlyweight(repeatingState);

  void operation() {
    _flyweight.operation(_uniqueState);
  }

  bool hasFlyweight(Flyweight flyweight) {
    return _flyweight == flyweight;
  }

  @override
  String toString() {
    return 'Context: uniqueState = $_uniqueState $_flyweight';
  }
}
